src/qt/bitcoin.cpp

```diff
@@ -188,24 +212,85 @@ int main(int argc, char *argv[])
 
     SetupEnvironment();
 
-    // Set default value to exit properly. Exit code 42 will trigger restart of the wallet.
-    int currentExitCode = 0;
- 
+    // Do this early as we don't want to bother initializing if we are just calling IPC
+    ipcScanRelay(argc, argv);
+
+    // Command-line options take precedence:
+    // Before this would of been done in main then config file loaded.
+    // We will load config file here as well.
+    ParseParameters(argc, argv);
+
+    ReadConfigFile(mapArgs, mapMultiArgs);
+
+    // Here we do it if it was started with the snapshot argument and we not TestNet
+    if (mapArgs.count("-snapshotdownload") && !mapArgs.count("-testnet"))
+    {
+        Upgrade Snapshot;
+
+        // Let's check make sure gridcoin is not already running in the data directory.
+        if (!LockDirectory(GetDataDir(), ".lock", false))
+        {
+            fprintf(stderr, "Cannot obtain a lock on data directory %s.  Gridcoin is probably already running.", GetDataDir().string().c_str());
+
+            exit(1);
+        }
+
+        else
+            Snapshot.SnapshotMain();
+
+        // Delete snapshot regardless of result.
+        Snapshot.DeleteSnapshot();
```

Comment:
This will not delete the snapshot.zip file if SnapshotMain() throws an exception.